dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/communityManager/accessRequests/AccessToActiveMatchingDatasetTest.kt

```diff
@@ -0,0 +1,187 @@
+    }
+
+    @Test
+    fun privateFrameworkHasNoAccessNoMatchingDataset() {
+        jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.PremiumUser)
+
+        val singleDataRequest = vsmeUtils.setSingleDataVSMERequest(
+            apiAccessor.uploadOneCompanyWithRandomIdentifier()
+                .actualStoredCompany.companyId,
+            setOf("2022"),
+        )
+
+        requestControllerApi.postSingleDataRequest(singleDataRequest)
+        Thread.sleep(timeSleep)
+        // val recentReaderDataRequest = requestControllerApi.getDataRequestsForRequestingUser()[0]
+        val recentReaderDataRequest = requestControllerApi.getDataRequestsForRequestingUser().maxByOrNull {
+            it.creationTimestamp
+        }
+        // todo variable aus den beiden unteren Zeilen
+        assertEquals(AccessStatus.Pending, recentReaderDataRequest?.accessStatus)
+        assertEquals(RequestStatus.Open, recentReaderDataRequest?.requestStatus)
+    }
+
+    @Test
+    fun privateFrameworkHasNoAccessHasMatchingDataset() {
+        companyId = apiAccessor.uploadOneCompanyWithRandomIdentifier().actualStoredCompany.companyId
+        createVSMEDataAndPostAsAdminCompanyOwner(companyId)
+
+        jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.PremiumUser)
+
+        val singleDataRequest = vsmeUtils.setSingleDataVSMERequest(companyId, setOf("2022"))
+        requestControllerApi.postSingleDataRequest(singleDataRequest)
+        // TODO Maybe find different solution to Thread.sleep
+        Thread.sleep(timeSleep)
+        val recentReaderDataRequest = requestControllerApi.getDataRequestsForRequestingUser().maxByOrNull {
+            it.creationTimestamp
+        } //
+
+        assertEquals(AccessStatus.Pending, recentReaderDataRequest?.accessStatus)
+        assertEquals(RequestStatus.Answered, recentReaderDataRequest?.requestStatus)
+    }
+
+    @Test
+    fun privateFrameworkCompanyOwnerPrivateRequestAnswer() {
+        jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.PremiumUser)
+        companyId = apiAccessor.uploadOneCompanyWithRandomIdentifier().actualStoredCompany.companyId
+
+        val singleDataRequest = vsmeUtils.setSingleDataVSMERequest(companyId, setOf("2022"))
+
+        requestControllerApi.postSingleDataRequest(singleDataRequest)
+        Thread.sleep(timeSleep)
+        // var recentReaderDataRequest = requestControllerApi.getDataRequestsForRequestingUser()[0]
+        var recentReaderDataRequest = requestControllerApi.getDataRequestsForRequestingUser().maxByOrNull {
+            it.creationTimestamp
+        }
+
+        assertEquals(AccessStatus.Pending, recentReaderDataRequest?.accessStatus)
+        assertEquals(RequestStatus.Open, recentReaderDataRequest?.requestStatus)
+
+        createVSMEDataAndPostAsAdminCompanyOwner(companyId)
+
+        jwtHelper.authenticateApiCallsWithJwtForTechnicalUser(TechnicalUser.PremiumUser)
```

Comment:
this looks like a duplicate of the code above. please export into separate function

also: aren't you still authenticated as PremiumUser here?

Reply:
The authentication as PremiumUser has been exported into the 'createVSMEDataAndPostAsAdminCompanyOwner' function. In the function 'createVSMEDataAndPostAsAdminCompanyOwner' the user is changed to admin and thus has to be changed back to PremiumUser for the following steps of the test.